### ISSUE TITLE
KeyError: '\\details' for signature-translated R functions without details section in docs

### ISSUE THREAD

**Issue by krassowski** [rpy2/rpy2#902]
**Describe the issue or bug**

Signature-translated R functions which do not have "Details" section in documentation error--out on runtime generation of Python docstrings.

**To Reproduce**

```python
from rpy2.robjects.packages import importr
stats = importr('stats')
print(stats.rnorm.__doc__)  # works ok
print(stats.summary_aov.__doc__)   # KeyError: '\\details'
```

**Expected behavior**
Details section gets skipped if absent, best possible docstring is generated instead.

**Error**

```python
File rpy2/robjects/functions.py:55, in DocstringProperty.__get__(self, obj, objtype)
     53     return self.class_doc
     54 else:
---> 55     return self.fget(obj)

File py2/robjects/functions.py:251, in DocumentedSTFunction.__doc__(self)
    248     doc.append(' '.join(('  ', item.name, ': ', description)))
    249     doc.append('')
--> 251 doc.append(help.docstring(self.__rpackagename__,
    252                           self.__rname__,
    253                           sections=['\\details'])
    254 )
    256 return os.linesep.join(doc)

File rpy2/robjects/help.py:426, in docstring(package, alias, sections)
    424     package = Package(package)
    425 page = package.fetch(alias)
--> 426 return page.to_docstring(sections)

File rpy2/robjects/help.py:293, in Page.to_docstring(self, section_names)
    291 s.append(os.linesep)
    292 s.append(os.linesep)
--> 293 walk(self.sections[name])
    294 s.append(os.linesep)
    295 s.append(os.linesep)

KeyError: '\\details'
```

**Additional context**

rpy2 3.5.2

**Comment by krassowski** [comment]
Here is a handy test script to check for `__doc__` issues for base package functions:

```python
key_errors = {}
lookup_errors = {}
other_errors = {}
for name in tqdm(dir(base)):
    value = getattr(base, name)
    if isinstance(value, SignatureTranslatedFunction):
        try:
            value.__doc__
        except KeyError as e:
            key_errors[name] = e
        except LookupError as e:
            lookup_errors[name] = e
        except Exception as e:
            other_errors[name] = e
```

Result:
- there are 84 functions without `'\\details'` section in `base` alone; this is even more common in custom libraries where authors may not be as thorough with documentation.
- there are 47 functions for which docstring generation fails on an earlier step with `LookupError('names')`, including `getwd` and `Sys_Date` (`Sys.Date`); I will open another issue.
- no other exceptions in `base` or `stats`

it takes 2 minutes to run, so I am not 100% sure about including this as a test case, though this would be very neat way of ensuring that docs strings for all base functions at least do not error out. Any preferences?

**Comment by lgautier** [comment]
> it takes 2 minutes to run, so I am not 100% sure about including this as a test case, though this would be very neat way of ensuring that docs strings for all base functions at least do not error out. Any preferences?

Thanks. Having the script somewhere in the source tree would be handy (may be in `scripts/` or in `doc/_static/demos/`). I am reluctant to add a long-running tests that runs with every commit though (the current setting for automated testing). A suite of extensive tests that run when preparing a release would make more sense.

I see that you already have a PR. I'll review it shortly.

**Comment by lgautier** [comment]
Fixed with PR #907.
